### PR TITLE
Adding metrics for apiserver request count.

### PR DESCRIPTION
**Pull request by krzysied**

### CODE REVIEW

File: perfdash/config.go [hunk 1]
```diff
@@ -79,6 +79,11 @@ var (
 			OutputFilePrefix: "APIResponsiveness",
 			Parser:           parseRequestCountData,
 		},
+		"DensityApiserverRequestCount": {
```
Comment (shyamjvs) [inline perfdash/config.go]: `DensityRequestCount` should be enough. Same below.

Comment (shyamjvs) [reply perfdash/config.go]: Ok, seems like that's already taken. How about `DensityRequestCountByClient`?

File: perfdash/parser.go [hunk 2]
```diff
@@ -137,3 +137,58 @@ func parseRequestCountData(data []byte, buildNumber int, testResult *BuildData)
 		}
 	}
 }
+
+type metricValue struct {
```
Comment (shyamjvs) [inline perfdash/parser.go]: You probably shouldn't need to redefine a struct here. It should already be present somewhere in kubernetes/kubernetes code. Just try to import that package.

File: perfdash/parser.go [hunk 3]
```diff
@@ -137,3 +137,58 @@ func parseRequestCountData(data []byte, buildNumber int, testResult *BuildData)
...
+	Metric	map[string]string	`json: metric`
+	Value	[]interface{}		`json: value`
+}
+
+type e2eMeteric []metricValue
+
+type e2eMetrics struct {
+	ApiServerMetrics		map[string]e2eMeteric	`ApiServerMetrics`
+	ClusterAutoscalerMetrics	map[string]e2eMeteric	`ClusterAutoscalerMetrics`
```
Comment (shyamjvs) [inline perfdash/parser.go]: Why do you need this?

File: perfdash/config.go (hunk 1 above)
Comment (krzysied) [reply perfdash/config.go]: Done

File: perfdash/parser.go (hunk 2 above)
Comment (krzysied) [reply perfdash/parser.go]: I will add import, however I want to do this in a separate pull request. It requires changes in godep dependencies.

Comment (shyamjvs) [reply perfdash/parser.go]: sgtm

File: perfdash/parser.go [hunk 4]
```diff
@@ -137,3 +137,58 @@ func parseRequestCountData(data []byte, buildNumber int, testResult *BuildData)
...
+type metricValue struct {
+	Metric map[string]string `json: metric`
+	Value  []interface{}     `json: value`
+}
+
+type e2eMeteric []metricValue
+
+type e2eMetrics struct {
+	ApiServerMetrics         map[string]e2eMeteric `ApiServerMetrics`
+	ClusterAutoscalerMetrics map[string]e2eMeteric `ClusterAutoscalerMetrics`
```
Comment (shyamjvs) [inline perfdash/parser.go]: Can we somehow get rid of this? It looks strange that we're specifically also getting ClusterAutoscalerMetrics.

File: perfdash/parser.go [hunk 5]
```diff
@@ -137,3 +137,58 @@ func parseRequestCountData(data []byte, buildNumber int, testResult *BuildData)
...
+}
+
+func createApiserverRequestCountData(data *perftype.DataItem, metric *metricValue) error {
+	data.Unit = ""
+	data.Data = make(map[string]float64)
+	data.Labels = metric.Metric
+	delete(data.Labels, "__name__")
```
Comment (shyamjvs) [inline perfdash/parser.go]: I think you probably can also get rid of `contentType` label - it's not too useful IMO.

File: perfdash/parser.go (hunk 2 above)
Comment (shyamjvs) [reply perfdash/parser.go]: Can you please add a TODO above the struct saying that we should eventually replace it with the dependency?

**Comment by shyamjvs** [comment]
Have few minor comments - looks great overall.
Thanks!

File: perfdash/parser.go (hunk 5 above)
Comment (krzysied) [reply perfdash/parser.go]: done

File: perfdash/parser.go (hunk 3 above)
Comment (krzysied) [reply perfdash/parser.go]: ClusterAutoscalerMetrics removed

File: perfdash/parser.go (hunk 2 above)
Comment (krzysied) [reply perfdash/parser.go]: done

File: perfdash/parser.go [hunk 6]
```diff
@@ -137,3 +137,59 @@ func parseRequestCountData(data []byte, buildNumber int, testResult *BuildData)
 		}
 	}
 }
+
+//TODO: replace this strucure with proper structure from metric dependecy.
```
[the remark this answers is not in the document]
Comment (krzysied) [reply perfdash/parser.go]: done

**Comment by shyamjvs** [comment]
/lgtm
Thanks.

**Pull request merged**